protocols/v2/roles-logic-sv2/src/utils.rs

```diff
@@ -563,6 +564,47 @@ pub fn get_target(
     hash
 }
 
+pub fn short_hash_list(
+    tx_data: Seq064K<'static, B016M<'static>>,
+    tx_short_hash_nonce: u64,
+) -> (Seq064K<'static, ShortTxId<'static>>, U256<'static>) {
+    let tx_short_hash_list = short_hash_list_builder(tx_short_hash_nonce, tx_data.clone());
+    let tx_hash_list_hash = tx_hash_list_hash_builder(tx_data);
+    (tx_short_hash_list, tx_hash_list_hash)
+}
+
+fn short_hash_list_builder(
+    tx_short_hash_nonce: u64,
+    tx_data: Seq064K<'static, B016M<'static>>,
+) -> Seq064K<'static, ShortTxId<'static>> {
+    // hash the short hash nonce
+    let nonce_hash = sha256::Hash::hash(&tx_short_hash_nonce.to_le_bytes());
+    // take first two integers from the hash
+    let k0 = u64::from_le_bytes(nonce_hash[0..8].try_into().unwrap());
+    let k1 = u64::from_le_bytes(nonce_hash[8..16].try_into().unwrap());
+    let mut tx_short_hash_list = vec![];
+    // get every transaction, hash it, remove first two bytes and push the ShortTxId in a vector
+    for tx in tx_data.to_vec() {
+        let hasher = SipHasher24::new_with_keys(k0, k1);
+        let tx_hashed = hasher.hash(&tx);
+        let tx_hashed_bytes: Vec<u8> = tx_hashed.to_le_bytes().to_vec().drain(0..2).collect();
+        let tx_hashed_bytes: ShortTxId = tx_hashed_bytes.try_into().unwrap();
+        tx_short_hash_list.push(tx_hashed_bytes);
+    }
+    Seq064K::from(tx_short_hash_list)
+}
+
+fn tx_hash_list_hash_builder(_tx_data: Seq064K<'static, B016M<'static>>) -> U256 {
+    // TODO: understand if this field is redunant and to be deleted since
```

Comment:
Yep for now is ok this solution cause we are not going to use in any way this field on our implementation. Btw we should open an issue for it.